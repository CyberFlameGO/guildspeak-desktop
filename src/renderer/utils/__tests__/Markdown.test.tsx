import React from 'react'
import { render, cleanup } from '@testing-library/react'
import richMarkdown from '../markdown'

afterEach(cleanup)

const md = `
**bold**
__bold2__
*italic*
_italic2_
plain text
`

describe('RichMarkdown component', () => {
  it('should render component', () => {
    render(richMarkdown(md))
  })

  it('should render proper html tag for bold', () => {
    const { getByText } = render(richMarkdown(md))
    expect(getByText('bold').tagName).toBe('STRONG')
    expect(getByText('bold2').tagName).toBe('STRONG')
  })

  it('should render proper html tag for italic', () => {
    const { getByText } = render(richMarkdown(md))
    expect(getByText('italic').tagName).toBe('EM')
    expect(getByText('italic2').tagName).toBe('EM')
  })

  it('should render proper html tags for bold and italic', () => {
    const { getByTestId } = render(
      <div data-testid="md">{richMarkdown(`This text is ***really important***.`)}</div>
    )

    expect(getByTestId('md').innerHTML).toBe(
      'This text is <strong><em>really important</em></strong>.'
    )
  })

  it('should render proper html tag for plain text', () => {
    const { getByText } = render(richMarkdown(md))
    expect(getByText('plain text').tagName).toBe('DIV')
  })

  it('should render div tag for link instead of `a`', () => {
    const { getByTestId } = render(
      <div data-testid="md">
        {richMarkdown(`My favorite website is [Example](https://example.com).`)}
      </div>
    )
    expect(getByTestId('md').innerHTML).toBe('My favorite website is Example.')
  })
})
