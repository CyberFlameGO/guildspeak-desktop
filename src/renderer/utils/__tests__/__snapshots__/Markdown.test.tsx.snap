// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichMarkdown component should render component 1`] = `
<DocumentFragment>
  <strong>
    bold
  </strong>
  <br />
  <strong>
    bold2
  </strong>
  <br />
  <em>
    italic
  </em>
  <br />
  <em>
    italic2
  </em>
  <br />
  plain text
</DocumentFragment>
`;
